src/Compression/Compression.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using ObjCRuntime;
+
+namespace Compression {
+
+
+	public partial class CompressionStream : IDisposable {
+
+		[StructLayout(LayoutKind.Sequential)]
+		struct CompressionStreamStruct {
+			public IntPtr Destination;
+			public int DestinationLength;
+			public IntPtr Source;
+			public int SourceLength;
+			public IntPtr State;
+		}
+
+		[DllImport (Constants.libcompression)]
+		static extern int /* CompressionStatus */ compression_stream_init (ref CompressionStreamStruct stream, int /* StreamOperation */ operation, int /* CompressionAlgorithm */ algorithm);
```

Comment:
The signature can use `CompressionAlgorithm` instead of `int` and the cast in the code